cmd/manager/main.go

```diff
@@ -59,7 +60,15 @@ func main() {
 	}
 
 	// Setup a Manager
-	mgr, err := manager.New(cfg, manager.Options{MetricsBindAddress: *metricsAddr})
+	opts := manager.Options{
+		MetricsBindAddress: *metricsAddr,
+	}
+	if *watchNamespace != "" {
+		opts.Namespace = *watchNamespace
+		klog.Infof("Watching machine-api objects only in namespace %q for reconciliation.", opts.Namespace)
+	}
+
```

Comment:
This doesn't seem OpenShift specific, so let's put it in the upstream version first

Reply:
https://github.com/metal3-io/cluster-api-provider-baremetal/pull/89